Write lens report format version to report and validate it on load

zblz opened the pull request:
This introduces a `_version` field in the report that indicates the version of the report format.

Eventually, the report will have a stable version and not introduce breaking changes, but for now the format version is set to be the same as the lens version, and validated to be equal on loading the report.

Fixes #9.

acroz commented:
Could this be `_schema_version` or something equivalent rather than `_version`? `_version` suggests a version of a report rather than the report format to me.

zblz commented:
I've changed the key to be `_schema_report` as @acroz suggested, and moved it into the Summary class as an integer as opposed to using lens' version. Even if the format is not considered stable, I think it will be useful to have that as an indicator of changes to the format, as requiring the same lens version on generation and loading might be too restrictive.

srstevenson commented:
@zblz Is this ready to merge?

zblz commented:
I changed the PR significantly since you approved it by moving the key into the summary (see message above). I forgot to ping you to re-review. Can you take a look at current changes and confirm you are OK with it?

srstevenson commented:
@zblz LGTM!
